src/tcp.c

```diff
@@ -188,8 +188,11 @@ get_if_ip(char *iface, char *ip)
 	if (ioctl(fd, SIOCGIFADDR, &ifr) == 0) {
```

Comment:
You could instead do:

	...
	int ret = !ioctl(fd, SIOCGIFADDR, &ifr);
	if (ret) {
		...
	}
	close(fd);
	return ret;

Reply:
yeah. can be re-arranged, but I wanted to keep the change in the scope of the bugfix (restyle could come later). But let me re-arrange that now..